misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperCodeActionUtil.java

```diff
@@ -282,41 +268,35 @@ public static AIDataMapperCodeActionUtil getInstance() {
                             }
                         }
                         if (functionCall == null) {
-                            return fEdits;
+                            return outputArray;
                         }
                         symbolAtCursorName = functionCall.trim();
                         generatedFunctionName = String.format("map%sTo%s(%s)",
                                 foundTypeRight, foundTypeLeft, symbolAtCursorName);
-                        fEdits.add(new TextEdit(newTextRange, generatedFunctionName));
                     } else {
                         symbolAtCursorName = functionCall.trim();
                         generatedFunctionName =
                                 String.format("map%sTo%s(%s)", foundTypeRight, foundTypeLeft, symbolAtCursorName);
-                        fEdits.add(new TextEdit(newTextRange, generatedFunctionName));
                     }
                     break;
                 default:
-                    return fEdits;
+                    return outputArray;
             }
 
             // Insert function declaration at the bottom of the file
             String functionName = String.format("map%sTo%s", foundTypeRight, foundTypeLeft);
             boolean found = fileContentSymbols.stream().anyMatch(p -> p.getName().get().contains(functionName));
-            if (found) {
-                return fEdits;
-            } else {
-                // Get the last line of the file
+            if (!found) {
                 syntaxTree = context.workspace().syntaxTree(context.filePath()).get();
-                TextDocument fileContentTextDocument = syntaxTree.textDocument();
-                int numberOfLinesInFile = fileContentTextDocument.toString().split("\n").length;
-                Position startPosOfLastLine = new Position(numberOfLinesInFile + 2, 0);
-                Position endPosOfLastLine = new Position(numberOfLinesInFile + 2, 1);
-                Range newFunctionRange = new Range(startPosOfLastLine, endPosOfLastLine);
-
-                // Get the generated record mapping function
-                String mappingFromServer =
-                        getGeneratedRecordMapping(context, foundTypeLeft, foundTypeRight, lftTypeSymbol,
+
+                JsonArray schemas =
+                        getGeneratedRecordMapping(foundTypeLeft, foundTypeRight, lftTypeSymbol,
                                 rhsTypeSymbol, syntaxTree, semanticModel);
+                String url = LSClientConfigHolder.getInstance(context.languageServercontext())
+                        .getConfigAs(ClientExtendedConfigImpl.class).getDataMapper()
+                        .getUrl() + "/map/2.0.0";
```

Comment:
Is hardcoding the path postfix valid?

Reply:
This is an internal versioned path of the data mapper service. That is why this is hard corded.